logic/src/commonMain/kotlin/com/wire/kalium/logic/data/connection/ConnectionRepository.kt

```diff
@@ -139,14 +153,38 @@ internal class ConnectionDataSource(
         // This can fail? but the connection will be there and get synced in worst case in next SlowSync
         wrapApiRequest {
             userDetailsApi.getUserInfo(idMapper.toApiModel(connection.qualifiedToId))
-        }.flatMap {
+        }.flatMap { userProfileDTO ->
             wrapStorageRequest {
-                val userEntity = publicUserMapper.fromUserApiToEntity(it, connectionStatusMapper.toDaoModel(connection.status))
+                val userEntity = publicUserMapper.fromUserApiToEntityWithConnectionStateAndUserTypeEntity(
+                    userDetailResponse = userProfileDTO,
+                    connectionState = connectionStatusMapper.toDaoModel(state = connection.status),
+                    userTypeEntity = userTypeEntityTypeMapper.fromOtherUserTeamAndDomain(
+                        otherUserDomain = userProfileDTO.id.domain,
+                        selfUserTeamId = getSelfUser().teamId,
+                        otherUserTeamID = userProfileDTO.teamId
+                    )
+                )
+
                 userDAO.insertUser(userEntity)
             }
         }
     }
 
+    //TODO: code duplication here for getting self user, the same is done inside
+    // UserRepository, what would be best ?
+    // creating SelfUserDao managing the UserEntity corresponding to SelfUser ?
+    private suspend fun getSelfUser(): SelfUser {
```

Comment:
Yup I think the same, that we should have separate place where we store current user data